lib/build.js

```diff
@@ -656,7 +657,7 @@ class BuildModel extends BaseModel {
             })
             .then(steps => {
                 if (!steps.length) {
-                    throw new Error('Steps do not exist');
```

Comment:
This error will be handled [this line](https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/get.js#L49) and a raw Error object becomes 500 response by `boom.boomify` method.
Therefore the object made by `boom.notFound` should be thrown. (and `boom.notFound()` will still be 404 response.)